### DISCUSSION TITLE
Using later version of nntool than in the aideck docker

### DISCUSSION THREAD

**Discussion by Jool99** [bitcraze/discussions#1237] [Q&A]
Problem:

* onnx model gets through nntool and autotiler with native install of gap_sdk version 4.22.0 but not with the ai deck docker.

* When trying to do full build with native install of gap_sdk, I get the following:
ap_sdk/tools/autotiler_v3/Emulation/at_api_pmsis.h:206:32: error: storage size of 'default_flash_conf' isn't known
struct pi_default_flash_conf default_flash_conf;

Questions:

What is the way forward here? Trying to use native SDK and provide something missing or to use the ai deck docker and adjust the model? Or is it possible to upgrade the autotiler inside the docker?

All help is appreciated. Thanks in advance!

**Comment by gemenerik** [comment]
Your two best options are; either redesign your model to be supported by the outdated gap_sdk in the docker, or flash with native gap_sdk. 

If you have issues with the gap_sdk like the one above, report them at the [gap_sdk repository](https://github.com/GreenWaves-Technologies/gap_sdk/issues).

Updating the sdk inside the docker is not straightforward so I do not recommend.
